op-devstack/sysgo/deployer.go

```diff
@@ -258,6 +258,34 @@ func WithInteropAtGenesis() DeployerOption {
 	}
 }
 
+// WithHardforkSequentialActivation configures a deployment such that L2 chains
+// activate hardforks sequentially, starting from startFork and continuing
+// until (but not including) endFork. Each successive fork is scheduled at
+// an increasing offset.
+func WithHardforkSequentialActivation(startFork, endFork rollup.ForkName, delta *uint64) DeployerOption {
+	return func(p devtest.P, keys devkeys.Keys, builder intentbuilder.Builder) {
+		for _, l2Cfg := range builder.L2s() {
+			l2Cfg.WithForkAtGenesis(startFork)
+			activateWithOffset := false
+			deactivate := false
+			for idx, refFork := range rollup.AllForks {
+				if deactivate || refFork == endFork {
+					l2Cfg.WithForkAtOffset(refFork, nil)
+					deactivate = true
+					continue
+				}
```

Comment:
Nit: We could just remove `deactivate` and have a `break` statement, no? I think that'd be simpler.

Reply:
We need to explicitly set the deactivation by `WithForkAtOffset(refFork, nil)`. Otherwise, all non-touched hardforks will set to genesis, causing below error:
```log
failed to combine L2 init config: combined deploy config failed validation: config field L2InitializationConfig failed checks: config field UpgradeScheduleDeployConfig failed checks: fork jovian set to 0, but prior fork isthmus has higher offset 120
```